WordPress/Classes/ViewRelated/NUX/SignupEpilogueTableViewController.swift

```diff
@@ -0,0 +1,241 @@
+
+            return cell
+        }
+
+        return nil
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+
+        if indexPath.section == TableSections.userInfo {
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: CellIdentifiers.epilogueUserInfoCell) as? EpilogueUserInfoCell else {
+                fatalError("Failed to get a user info cell")
+            }
+
+            if let epilogueUserInfo = epilogueUserInfo {
+                cell.configure(userInfo: epilogueUserInfo, showEmail: true)
+            }
+            userInfoCell = cell
+            return cell
+        }
+
+        if indexPath.section == TableSections.names {
+            if indexPath.row == 0 {
+                return getEpilogueCellFor(cellType: .displayName)
+            }
+
+            if indexPath.row == 1 {
+                return getEpilogueCellFor(cellType: .username)
+            }
+        }
+
+        if indexPath.section == TableSections.password {
+            return getEpilogueCellFor(cellType: .password)
+        }
+
+        return super.tableView(tableView, cellForRowAt: indexPath)
+    }
+
+    override func tableView(_ tableView: UITableView, estimatedHeightForHeaderInSection section: Int) -> CGFloat {
+        return Constants.headerFooterHeight
+    }
+
+    override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return UITableViewAutomaticDimension
+    }
+
+    override func tableView(_ tableView: UITableView, estimatedHeightForFooterInSection section: Int) -> CGFloat {
+        return Constants.headerFooterHeight
+    }
+
+    override func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        if section == TableSections.password {
+            return UITableViewAutomaticDimension
+        }
+        return 0
+    }
+
+}
+
+// MARK: - Private Extension
+
+private extension SignupEpilogueTableViewController {
+
+    func configureTable() {
+        let headerFooterNib = UINib(nibName: CellNibNames.sectionHeaderFooter, bundle: nil)
+        tableView.register(headerFooterNib, forHeaderFooterViewReuseIdentifier: CellIdentifiers.sectionHeaderFooter)
+
+        let cellNib = UINib(nibName: CellNibNames.signupEpilogueCell, bundle: nil)
+        tableView.register(cellNib, forCellReuseIdentifier: CellIdentifiers.signupEpilogueCell)
+
+        let userInfoNib = UINib(nibName: CellNibNames.epilogueUserInfoCell, bundle: nil)
+        tableView.register(userInfoNib, forCellReuseIdentifier: CellIdentifiers.epilogueUserInfoCell)
+
+        WPStyleGuide.configureColors(for: view, andTableView: tableView)
+
+        // remove empty cells
+        tableView.tableFooterView = UIView()
+    }
+
+    func getUserInfo() {
+        if loginFields.meta.socialService == .google {
+            showPassword = false
+            var userInfo = LoginEpilogueUserInfo()
```

Comment:
if `googleUser` was part of `LoginFields` then `LoginEpilogueUserInfo` could have a constructor like `init(fields: LoginFields)` here.